src/app/checkout/step-2/step-2.component.spec.js

```diff
@@ -308,6 +308,77 @@ describe('checkout', () => {
         expect(self.controller.paymentFormState).toBe('success')
         expect(self.controller.getContinueDisabled()).toBe(false)
       })
+
+      it('should return true when cvv validity is false and card type is credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = false
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(true)
+      })
+
+      it('should return true when cvv validity is true and card type is credit card', () => {
```

Comment:
I think a more natural language on some of these tests would be nicer. Instead of `cvv validity is true`, `cvv is valid` is more natural. This way of phrasing works fine for `undefined`, but I think we should be more natural on the true/false options.